_posts/2020-11-23-boost-your-productivity-with-tailwind-css.md

```diff
@@ -0,0 +1,184 @@
+></iframe>
+
+In example 1, we see Tailwind in its most basic form. It provides the button styles to make it look like a button. When the input is `class="p-2 leading-none text-white bg-purple-600 rounded-sm"` the output is
+
+```css
+.p-2 {
+  padding: 0.5rem;
+}
+.bg-purple-600 {
+  --bg-opacity: 1;
+  background-color: #805ad5;
+  background-color: rgba(128, 90, 213, var(--bg-opacity));
+}
+.leading-none {
+  line-height: 1;
+}
+.rounded-sm {
+  border-radius: 0.125rem;
+}
+.text-white {
+  --text-opacity: 1;
+  color: #fff;
+  color: rgba(255, 255, 255, var(--text-opacity));
+}
+```
+
+Typically you would give the element a single class, such as `class="btn"`, and then you would write in your CSS `.btn { ... }`. This example I've just introduced you is, of course, wildly exaggerated, but I want you to pay attention to the markup.
+
+```html
+<button class="p-2 leading-none text-white bg-purple-600 rounded-sm">Continue</button>
+<button class="btn">Continue</button>
+```
+
+Notice how it is immediately clearer how the button will look like just by looking at the markup. Utility-first CSS is all about these classes, how they are named, their output, and the connection between them. The classes are all reusable little snippets that provide functional styles, and since they already exist, you don't have to spend any effort to come up with fitting names.
+
+Using utility classes is like baking cupcakes; The element or component is the batter, and utility classes are the sprinkles on top. Sprinkles make the cupcake nice and shiny, while solid batter provides base and structure.
+
+![Cupcake sprinkling](/img/boost-your-productivity-with-tailwind-css/cupcake-sprinkling.jpg)
+
+Let's continue and take example 1 just a tiny step further in example 2.
+
+### Example 2 - Responsive prefixes
+
+<iframe
+  src="https://codesandbox.io/embed/responsive-prefixes-boost-your-productivity-with-tailwind-css-1hcks?codemirror=1&fontsize=14&hidenavigation=1&module=%2Fsrc%2FApp.js&theme=dark&runonclick=1&view=split&editorsize=66&highlights=10,11,12"
+  style="width:100%; height:500px; border:0; border-radius: 4px; overflow:hidden;"
+  title="Responsive prefixes | Boost your productivity with Tailwind CSS"
+  allow="accelerometer; ambient-light-sensor; camera; encrypted-media; geolocation; gyroscope; hid; microphone; midi; payment; usb; vr; xr-spatial-tracking"
+  sandbox="allow-forms allow-modals allow-popups allow-presentation allow-same-origin allow-scripts"
+></iframe>
+
+One of the most crucial aspects of designing for the web today is responsiveness. Tailwind uses many prefixes for class names, and the most apparent of them are responsive utilities: `sm`, `md`, `lg`, and `xl`. By default, you can use these prefixes with any other class. You can even use them together with different prefixes like this `sm:hover:bg-pink-700`. You can, of course, customize the default breakpoints and even add your own in `tailwind.config.js`.
```

Comment:
”One of the most crucial aspects of designing for the web today is responsiveness.”

Even better than Responsive is Fluid or Elastic! I guess my favourite is mostly-fluid/elastic with some added responsiveness and fixed elements for edge cases. I never really liked responsive designs.

Does Tailwind ”support” other design styles than Responsive? Maybe you might want to add something to say why Responsive is a good thing, why to not use other styles or how to use them with Tailwind?

Reply:
I'm not quite sure what you mean by fluid and elastic. Maybe responsive design would be better in this context. I think of it as an umbrella term that includes "standard" responsive, fluid and whatnot.

Reply:
Ah, you are probably right, I guess responsive design is already an umbrella term, or at least something like a combination of breakpoints and fluidity between them. What I had in mind was like only breakpoints without fluidity which I guess is closer to adaptive design. So my bad.